### PR TITLE
Separate stats updates from decision_maker

### PR DESCRIPTION
**Pull request by apfitzge**
#### Problem
Central scheduler will not use LeaderSlotReceiveMetricsTracker. 

#### Summary of Changes
Separate the updating of leader slot metrics from decision making so that scheduler isn't forced to use it.

Fixes #

### CODE REVIEW

File: core/src/banking_stage.rs [hunk 1]
```diff
@@ -433,8 +433,8 @@ impl BankingStage {
         if unprocessed_transaction_storage.should_not_process() {
             return;
         }
-        let ((metrics_action, decision), make_decision_time) =
-            measure!(decision_maker.make_consume_or_forward_decision(slot_metrics_tracker));
+        let (decision, make_decision_time) =
+            measure!(decision_maker.make_consume_or_forward_decision());
```
Comment (ryoqun) [inline core/src/banking_stage.rs:437]: hmm, `.check_leader_slot_boundary()` won't be called for the case of `BufferedPacketsDecision::Hold`. So, technically this is breaking change with regard to the precise definition of those metrics. (as i ranted earlier at banking trace pr, i wonder this is worth to retain so insistently....)

Anyway, I'd suggest to call `.check_leader_slot_boundary()` right here mainly for simplicity and maintenance ease:

```rust
let (decision, make_decision_time) =
    measure!(decision_maker.make_consume_or_forward_decision());
let metrics_action = slot_metrics_tracker.check_leader_slot_boundary(decision.bank_start());


...


impl BufferedPacketDecision {
  fn bank_start(&self) -> &Option<BankStart> {
       match self { ... }
  }
}

```

**Review by ryoqun — COMMENTED** [review]
sorry for delay. just commented some.

File: core/src/banking_stage.rs (hunk 1 above)
Comment (apfitzge) [reply core/src/banking_stage.rs:437]: I think it's likely better to keep the type juggling change - `bank_start(&self) -> &Option<BankStart>` would require cloning the `BankStart` (2 `Arc`s). But we can just copy the reference into an option, and avoid those clones 😄

File: core/src/banking_stage/decision_maker.rs [hunk 2]
```diff
@@ -19,6 +18,15 @@ pub enum BufferedPacketsDecision {
     Hold,
 }
 
+impl BufferedPacketsDecision {
+    pub fn bank_start(&self) -> Option<&BankStart> {
+        match self {
+            Self::Consume(bank_start) => Some(bank_start),
```
Comment (apfitzge) [inline core/src/banking_stage/decision_maker.rs:24]: If we are returning `&Option<BankStart>`, I think we'd need to have
```
            Self::Consume(bank_start) => &Some(bank_start.clone()),
```

File: core/src/leader_slot_banking_stage_metrics.rs [hunk 3]
```diff
@@ -981,13 +981,13 @@ mod tests {
         {
             // Setup with first_bank
             let action = leader_slot_metrics_tracker
-                .check_leader_slot_boundary(&Some(first_poh_recorder_bank.clone()));
+                .check_leader_slot_boundary(Some(&first_poh_recorder_bank.clone()));
```
Comment (apfitzge) [inline core/src/leader_slot_banking_stage_metrics.rs:984]: this clone is not necessary anymore - surprised clippy didn't yell at me 😦

**Comment by apfitzge** [comment]
Rebased as I was hitting an unexpected sanity-check issue in a file I didn't touch 😦

**Review by ryoqun — APPROVED** [review]
lgtm! nice cleaning.

**Pull request merged**